shopping_mall/backend/app/services/shipping_tracker.py

```
@@ -38,14 +40,21 @@ def check_status(cls, shipment: Shipment) -> str:
         return new_status
 
     @classmethod
-    def update_shipment(cls, shipment: Shipment) -> bool:
-        """Update a single shipment. Returns True if status changed."""
+    def update_shipment(cls, shipment: Shipment, db: Optional[Session] = None) -> bool:
```

Ruff (0.15.12)

[error] Undefined name `Optional`

(F821)